src/components/TechSpark/Login.tsx

```diff
@@ -1,59 +1,61 @@
 import "./Login.css"
-import React, { Component, useEffect, useLayoutEffect, useRef } from 'react';
+import React, { useLayoutEffect, useRef } from 'react';
 import axios from "axios";
 
 const dataRoot = "https://techstartbackend.herokuapp.com"
 
-function Login(props) {
+function Login(props: any) {
 
     const [tracker, setTracker] = React.useState(false)
-    const [invalidUsername, setInvalidUsername] = React.useState(false)
-    const [invalidPassword, setInvalidPassword] = React.useState(false)
+    // const [invalidUsername, setInvalidUsername] = React.useState(false)
+    // const [invalidPassword, setInvalidPassword] = React.useState(false)
     const [invalidLogin, setInvalidLogin] = React.useState(false)
+    const [message, setMessage] = React.useState("")
     const firstUpdate = useRef(true);
     useLayoutEffect(() => {
+        let username = document.getElementById('loginUsername')
```

Comment:
I don't like keeping track of the username/password this way. let's useState for the username & password. 

ex. something like 

const [name, setName] = useState("")
<input value={name} onChange={(e)=>{e.preventDefault(); setName(e.target.value)}}... />

you'll need to strong type e (the event). I think it's InputEvent or something like that. If you search it up you should be able to find it.

Reply:
You could also use useReducer to handle related states like username and password